orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -98,60 +125,49 @@ def __init__(self, module):
 
     def cpu(self: T) -> T:
         '''Thin layer to capture device for ORTModule IO bindings'''
-        if self._device != 'cpu':
+
+        if self._device.type != 'cpu':
             self._require_export = True
-            self._device = 'cpu'
+            self._device = torch.device('cpu')
+
         return super(ORTModule, self).cpu()
 
-    def cuda(self: T, device: Optional[Union[int, device]] = None) -> T:
+    def cuda(self: T, device: Optional[Union[int, torch.device]] = None) -> T:
         '''Thin layer to capture device for ORTModule IO bindings'''
-        if device:
-            device = str(device)
-        else:
-            device = 'cuda'
-        if self._device != str(device):
+
+        if device is None and _get_device_str(self._device) != _get_default_device_str('cuda'):
             self._require_export = True
-            self._device = device
+            self._device = torch.device(_get_default_device_str('cuda'))
+        elif _get_device_str(self._device) != _get_device_str(device):
```

Comment:
If the ```device``` is ```None``` and we hit this ```elif``` block, our code would raise an exception: ```'Unsupported device type'```. Is that expected behavior?

Reply:
Nope. will break this expression in 2